File: exercism/java/bob/src/main/java/Bob.java
public class Bob {
    public String hey(String s) {

        if (s.isEmpty() || s.isBlank() || s.matches("\\s+")) {
            return "Fine. Be that way!";
        }

        if (s.matches("[A-Z]([a-z]|\\s|[0-9]|[,'])+\\!?")) {
            return "Whatever.";
        }

        if (s.trim().endsWith("!") || s.matches("([A-Z]|\\s|[0-9]|[%\\^\\*@#'\\$\\(\\)\\!])+\\!?")) {
            return "Whoa, chill out!";
        }

        if (s.matches("([A-Z]|\\s)+\\??")) {
            return "Calm down, I know what I'm doing!";
        }

        if (s.trim().endsWith("?")) {
            return "Sure.";
        }

        return "Whatever.";
    }
}

File: exercism/clojure/bob/src/bob.clj
(ns bob)

(defn response-for [s] 
  (cond
    (or (= 0 (count s)) (re-matches #"\s+" s)) "Fine. Be that way!"
    (re-matches #"([A-Z]|\s)+\?" s) "Calm down, I know what I'm doing!"
    (re-matches #"([1-9,]|\s)+" s) "Whatever."
    (re-matches #"([1-9A-Z]|\s|[,%\^\*@#\$\(\)\!])+\!?" s) "Whoa, chill out!"
    (re-matches #"([1-9a-zA-Z,]|\s)+\?" s) "Sure."
    :else "Whatever.")
)
